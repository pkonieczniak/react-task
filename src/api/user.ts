import { fetch } from "../utils";

interface User {
  id: number;
  name: string;
  username: string;
  email: string;
  address: {
    street: string;
    suite: string;
    city: string;
    zipcode: string;
    geo: {
      lat: string;
      lng: string;
    };
  };
  phone: string;
  website: string;
  company: {
    name: string;
    catchPhrase: string;
    bs: string;
  };
}

const resource = "users";

export const getUsers = fetch(
  `${process.env.REACT_APP_JSON_PLACEHOLDER_API_URL}${resource}`
);
