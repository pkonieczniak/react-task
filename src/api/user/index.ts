import { fetch } from "../../utils";
import { User } from "../../models/user";

const resource = "users";

export const getUsers = () => {
  return fetch<[User]>(
    `${process.env.REACT_APP_JSON_PLACEHOLDER_API_URL}${resource}`
  );
};
