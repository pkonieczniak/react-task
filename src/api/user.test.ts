// TODO: Improve mocking for the named imports, and clean up tests

jest.mock("../utils", () => {
  return {
    __esModule: true,
    fetch: jest.fn(() =>
      Promise.resolve([
        undefined,
        [
          { id: 1, name: "foo" },
          { id: 2, name: "bar" },
        ],
      ])
    ),
  };
});

import { getUsers } from "./user";

describe("getUsers", () => {
  it("returns the list of users", async () => {
    expect(await getUsers()).toMatchSnapshot();
  });
});
