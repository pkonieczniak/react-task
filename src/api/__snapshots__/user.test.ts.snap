// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getUsers returns the list of users 1`] = `
Array [
  undefined,
  Array [
    Object {
      "id": 1,
      "name": "foo",
    },
    Object {
      "id": 2,
      "name": "bar",
    },
  ],
]
`;
